src/stirling/source_connectors/socket_tracer/thriftmux_openssl_trace_bpf_test.cc

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2018- The Pixie Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <string>
+
+#include "src/common/base/base.h"
+#include "src/common/exec/exec.h"
+#include "src/common/testing/test_environment.h"
+#include "src/shared/types/column_wrapper.h"
+#include "src/shared/types/types.h"
+#include "src/stirling/core/data_table.h"
+#include "src/stirling/source_connectors/socket_tracer/socket_trace_connector.h"
+#include "src/stirling/source_connectors/socket_tracer/testing/container_images.h"
+#include "src/stirling/source_connectors/socket_tracer/testing/protocol_checkers.h"
+#include "src/stirling/source_connectors/socket_tracer/testing/socket_trace_bpf_test_fixture.h"
+#include "src/stirling/source_connectors/socket_tracer/uprobe_symaddrs.h"
+#include "src/stirling/testing/common.h"
+
+namespace px {
+namespace stirling {
+
+namespace mux = protocols::mux;
+
+using ::px::stirling::testing::EqHTTPRecord;
+using ::px::stirling::testing::EqMuxRecord;
+using ::px::stirling::testing::GetTargetRecords;
+using ::px::stirling::testing::SocketTraceBPFTestFixture;
+using ::px::stirling::testing::ToRecordVector;
+
+using ::testing::StrEq;
+using ::testing::UnorderedElementsAre;
+
+class ThriftMuxServerContainerWrapper : public ::px::stirling::testing::ThriftMuxServerContainer {};
+
+// The Init() function is used to set flags for the entire test.
+// We can't do this in the MuxTraceTest constructor, because it will be too late
+// (SocketTraceBPFTestFixture will already have been constructed).
+bool Init() {
+  // Make sure Mux tracing is enabled.
+  FLAGS_stirling_enable_mux_tracing = true;
```

Comment:
Will this cause us to run out of instructions when running the test on a 4.14 kernel?

Reply:
Nevermind...I see the code below. But what do you think of turning off other protocols so that the test can run on older kernels. Instead of just effectively disabling this test.

Reply:
Yep, that makes sense and I'll update that. I agree it would be great to have the test code exercised on all supported kernels.

Reply:
It seems that [mux_trace_bpf_test.cc](https://github.com/pixie-io/pixie/blob/d54536fc4aaeeebf027c17b47400d8ea4f4d6a45/src/stirling/source_connectors/socket_tracer/mux_trace_bpf_test.cc#L143-L150) is doing the same, so I'll update that as well.

Reply:
Updated.